### ISSUE TITLE
Inconsistent xstream-run signature while using TypeScript

### ISSUE THREAD

**Issue by geovanisouza92** [cyclejs/cyclejs#452]
I noticed a little inconsistency between xstream-run and most-run. Around cyclejs/cyclejs@2ca8e66, the signature changed from using DriversDefinition interface to an inline type.

You could see between [xstream-run](https://github.com/cyclejs/cyclejs/blob/5cb9d08e2db09bc14e9492fddb8bb3a73f3c1271/xstream-run/src/index.ts#L37) and [most-run](https://github.com/cyclejs/cyclejs/blob/5cb9d08e2db09bc14e9492fddb8bb3a73f3c1271/most-run/src/index.ts#L37). (Note: The latter is the intended signature.)

**Code to reproduce the issue:**

```typescript
import {run} from '@cycle/most-run'
import {makeDOMDriver, div, VNode} from '@cycle/dom'
import {DOMSource} from '@cycle/dom/most-typings'
import * as most from 'most'

type Sources = {
  DOM: DOMSource
}

type Sinks = {
  DOM: most.Stream<VNode>
}

function main (sources: Sources): Sinks {
  return {DOM: most.of(div('Hello'))}
}

const drivers = {
  DOM: makeDOMDriver('#app')
}

run(main, drivers)
```

**Expected behavior:**

It compiles without error messages.

**Actual behavior:**

While using TypeScript, I receive this error message:

```
$ node_modules/.bin/tsc

src/index.ts(22,11): error TS2345: Argument of type '{ DOM: Function; }' is not assignable to parameter of type 'DriversDefinition'.
  Property 'DOM' is incompatible with index signature.
    Type 'Function' is not assignable to type 'DriverFunction'.
      Type 'Function' provides no match for the signature '(stream: any, adapter: StreamAdapter, driverName: string): any'
```

To fix this, I'd to change the code like this:

`package.json`:

```diff
   "version": "1.0.0",
   "description": "",
   "dependencies": {
+    "@cycle/base": "^4.1.1",
     "@cycle/dom": "^12.2.6",
     "@cycle/most-run": "^4.1.1",
     "most": "^1.0.4",
```

`index.ts`:

```diff
import {run} from '@cycle/most-run'
import {makeDOMDriver} from '@cycle/dom'
import {App} from './app'
+import {DriverFunction} from '@cycle/base'

const main = App

const drivers = {
-  DOM: makeDOMDriver('#app')
+  DOM: makeDOMDriver('#app') as DriverFunction
}

run(main, drivers)
```

**Versions of packages used:**

```json
{
  "name": "example",
  "version": "1.0.0",
  "description": "",
  "dependencies": {
    "@cycle/dom": "^12.2.6",
    "@cycle/most-run": "^4.1.1",
    "most": "^1.0.4",
    "xstream": "^6.4.1"
  },
  "devDependencies": {
    "@types/core-js": "^0.9.34",
    "@types/node": "^6.0.45",
    "typescript": "^2.0.3"
  }
}
```

**Comment by staltz** [comment]
I can see the problem, and I understand that most-run is *supposedly* doing it right while xstream-run is doing it wrong. However, because we're not strictly checking the types in Cycle.run and because DriverFunction is a bit wrong (it assumes all driver functions have three arguments, which is not the case for quick one-liner drivers), I'll defer doing the right solution for later, probably together with issue #432.

For now, I'll change most-run typings to be like xstream-run.

**Comment by staltz** [comment]
most-run v4.1.2 released with this fix @geovanisouza92

**Comment by geovanisouza92** [comment]
@staltz Thanks, but, `import Cycle from '@cycle/most-run'` + `Cycle.run` works, while `import {run} from '@cycle/most-run'` does not. It seems that there is [one more change](https://github.com/cyclejs/cyclejs/blob/master/most-run/src/index.ts#L37) to be made in the signature to work.

The other *-run packages works fine.

**Comment by staltz** [comment]
Released v4.1.3 @geovanisouza92  :)
